services/movie_session.py

```diff
@@ -0,0 +1,40 @@
+from django.db.models import QuerySet
+from db.models import MovieSession
+import datetime
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int
+                         ) -> None:
+    MovieSession.objects.create(show_time=movie_show_time,
+                                cinema_hall_id=cinema_hall_id,
+                                movie_id=movie_id)
+
+
+def get_movies_sessions(session_date: datetime.date = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> QuerySet:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: datetime.time = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> None:
```

Comment:
Optional also

Reply:
it's about annotation type ? i don't understand

Reply:
Yes it is about the annotations